packages/fuselage-ui-kit/src/elements/ChannelsSelectElement/MultiChannelsSelectElement.spec.tsx

```diff
@@ -0,0 +1,132 @@
+import { MockedServerContext } from '@rocket.chat/mock-providers';
+import type { MultiChannelsSelectElement as MultiChannelsSelectElementType } from '@rocket.chat/ui-kit';
+import { BlockContext } from '@rocket.chat/ui-kit';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { contextualBarParser } from '../../surfaces';
+import MultiChannelsSelectElement from './MultiChannelsSelectElement';
+import { useChannelsData } from './hooks/useChannelsData';
+
+const channelsBlock: MultiChannelsSelectElementType = {
+  type: 'multi_channels_select',
+  appId: 'test',
+  blockId: 'test',
+  actionId: 'test',
+};
+
+jest.mock('./hooks/useChannelsData');
+
+const mockedOptions = [
+  {
+    value: 'channel1_id',
+    label: {
+      name: 'Channel 1',
+      avatarETag: 'test',
+      type: 'c',
+    },
+  },
+  {
+    value: 'channel2_id',
+    label: {
+      name: 'Channel 2',
+      avatarETag: 'test',
+      type: 'c',
+    },
+  },
+  {
+    value: 'channel3_id',
+    label: {
+      name: 'Channel 3',
+      avatarETag: 'test',
+      type: 'c',
+    },
+  },
+];
+
+const mockUseChannelsData = jest.mocked(useChannelsData);
+mockUseChannelsData.mockReturnValue(mockedOptions);
+
+describe('UiKit MultiChannelsSelect Element', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  beforeEach(() => {
+    render(
+      <MockedServerContext>
+        <MultiChannelsSelectElement
+          index={0}
+          block={channelsBlock}
+          context={BlockContext.FORM}
+          surfaceRenderer={contextualBarParser}
+        />
+      </MockedServerContext>
+    );
+  });
+
+  it('should render a UiKit multiple channels selector', async () => {
+    expect(await screen.findByRole('textbox')).toBeInTheDocument();
+  });
+
+  it('should open the channels selector', async () => {
+    const input = await screen.findByRole('textbox');
+    input.focus();
+
+    expect(await screen.findByRole('listbox')).toBeInTheDocument();
+  });
+
+  it('should filter channels', async () => {
+    const input = (await screen.findByRole('textbox')) as HTMLInputElement;
+
+    await userEvent.type(input, 'Channel 2', { delay: null });
+
+    mockUseChannelsData.mockReturnValueOnce(
```

Comment:
This here is a little concerning, because the filter behaviour is defined in the hook. By mocking the return value, it is not a real test of the filter.

IMO we can remove this filter test as it is not really testing anything. One option would be to mock the request response but since the filtering is done on the server, we'd still have the same problem (we're not really testing anything relevant).